### PR TITLE
WIP: Added files needed to create a Julia Package

### PR DESCRIPTION
**Pull request by lucylow**

### CODE REVIEW

**Comment by lucylow** [comment]
Still needs work

**Comment by Roger-luo** [comment]
Thanks for the PR. A few comments on this

- 1. you don't need to open another PR for this, just push to the original branch
- 2. if your PR is not ready for review you could add a prefix in the title as `WIP: `  which stands for work-in-progress, similarly, if you have something not sure, you could use the prefix `RFC` stands for request for comments
- 3. please remove those unnecessary files

File: REQUIRE [hunk 1]
```diff
@@ -0,0 +1,2 @@
+julia 0.5
```
Comment (Roger-luo) [inline REQUIRE]: no need to have this file.

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+environment:
```
Comment (Roger-luo) [inline appveyor.yml]: we are not gonna use appveyor

File: src/YaoPlots.jl [hunk 3]
```diff
@@ -1,7 +1,52 @@
+# YaoPlots.jl 
+#This is a component package module for [Yao.jl](https://github.com/QuantumBFS/Yao.jl)
+
+# User can run using YaoPlots using export plot function. Exports are the public API: internal functionality should not be export. Anything else must be qualified by  package name (YaoPlots.not_exported).ed!
+export Plot, output_string, added_function 
+
+
+__precompile__()
```
Comment (Roger-luo) [inline src/YaoPlots.jl]: no need to have precompile here. this is a very ancient thing.

File: src/output_string.jl [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# Internal file
```
Comment (Roger-luo) [inline src/output_string.jl]: this is an extra file as well?

Comment (Roger-luo) [reply src/output_string.jl]: and this is also ancient Julia grammar.

**Review by Roger-luo — COMMENTED** [review]
remove .DS_store and put this file to `.gitignore` and try to clean up your changes so it's easier to read.

File: docs/mkdocs.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# See the mkdocs user guide for more information on these settings.
```
Comment (Roger-luo) [inline docs/mkdocs.yml:1]: this file is not necessary.

File: .github/Julia.gitignore [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Files generated by invoking Julia with --code-coverage
```
Comment (Roger-luo) [inline .github/Julia.gitignore]: move this to the top most path, and rename to `.gitignore`

File: src/Catlab.jl [hunk 7]
```diff
@@ -0,0 +1,74 @@
+# Catlab.jl
```
Comment (Roger-luo) [inline src/Catlab.jl:1]: this is not something you should include in this PR, you could have it locally if you just wanna try it out.

File: src/GR.jl [hunk 8]
```diff
@@ -0,0 +1,5 @@
+plot(x::AbstractBlock; figsize=(10cm, 1cm), fontsize=10) = GR(context(0, 0, figsize...),
```
Comment (Roger-luo) [inline src/GR.jl:1]: can we have a working Compose based implementation first? Try not to commit everything in one single PR, it's not a good practice.

**Pull request closed without merging**